iOS/Inventory/DittoManager.swift

```diff
@@ -6,144 +6,242 @@
         } catch {
             let dittoErr = (error as? DittoSwiftError)?.errorDescription
             assertionFailure(dittoErr ?? error.localizedDescription)
         }
-
-        return ditto
     }
 
     var dittoInfoView: DittoInfoViewController {
         DittoInfoViewFactory.create(ditto: ditto)
     }
 }
 
-
 // MARK: - Upsert Methods
 
 extension DittoManager {
 
+    /// Subscribes to all inventory items in the "inventories" collection.
+    ///
+    /// This method registers a subscription to the "inventories" collection using a DQL query.
+    /// It also sets up an observer to monitor changes in the database, calculating the differences
+    /// between syncs using a DittoDiffer. The observer sends updates to the `itemsUpdated` subject
+    /// based on the type of change (initial load or update).
+    ///
+    /// - Note: This method should be called to start monitoring inventory items for changes.
     func subscribeAllInventoryItems() {
-
+        let query = "SELECT * FROM inventories"
         do {
-            subscriptions.append(try ditto.sync.registerSubscription(query: "SELECT * FROM inventories"))
-        } catch {
-            print("Query Error: \(error)")
-        }
-        
-        liveQueries.append(
-            collections.inventories.findAll().observeLocal { [weak self] docs, event in
-                guard let self = self else { return }
-
-                let allItems = docs.map { ItemDittoModel($0) }
-                self.models.items = allItems
-
-                switch event {
-                case .initial:
-
-                    self.itemsUpdated.send((indices: allItems.indexes, event: event))
-
-                case .update(let detail):
-
-                    self.itemsUpdated.send((indices: detail.updates, event: event))
-
-                @unknown default: break
+            // Create Subscription
+            // https://docs.ditto.live/sdk/latest/sync/syncing-data#creating-subscriptions
+            self.subscription = try ditto.sync.registerSubscription(
+                query: query
+            )
+
+            // DittoDiffer - used to calculate the delta changes between syncs
+            // TODO put in link to docs that describes the differ
+            let dittoDiffer = DittoDiffer()
+
+            // Register Observer to see changes in the database from sync
+            // https://docs.ditto.live/sdk/latest/crud/observing-data-changes
+            storeObserver = try ditto.store.registerObserver(query: query) {
+                [weak self] results in
+                
+                let allItems = results.items.compactMap{ ItemDittoModel($0.jsonString()) }
+                self?.models.items = allItems
+                let diff = dittoDiffer.diff(results.items)
+
+                // NOTE:  if you are curious on why we don't handle deletions - the app code
+                // currently does allow deleting of inventory items, so there is no reason to handle
```

Comment:
You mean `does not`?
```suggestion
                // currently does not allow deleting of inventory items, so there is no reason to handle
```